pyrene/src/utils/DateUtils.ts

```diff
@@ -0,0 +1,264 @@
+  const tStamp = new Date(date.year, date.month - 1, date.day, time.hours, time.minutes);
+  return tStamp.valueOf();
+};
+
+/**
+ * Provides the Date Object of current date/time
+ */
+export const getCurrentJsDateObject = (): Date => new Date();
+
+/**
+ * Provides the DateType object of current date/time
+ */
+export const getCurrentDateTypeObject = (): DateType => convertToDateTypeObject(new Date());
+
+/**
+ * Returns the Date Object of provided timestamp in provided timezone
+ */
+export const convertToZoneTime = (timestamp: number, timezone: string): Date => utcToZonedTime(timestamp, timezone);
+
+/**
+ * Returns the Date Object of provided timestamp in UTC
+ */
+export const convertToUTCtime = (datetime: string | number, timezone: string): Date => zonedTimeToUtc(datetime, timezone);
+
+/**
+ * Returns the timestamp of a point in time in the future relative to now
+ *
+ * @param {Duration} duration
+ */
+export const getFutureDate = (duration: Duration): number => add(getCurrentJsDateObject(), duration).valueOf();
+
+/**
+ * Returns the timestamp of a point in time in the past relative to now
+ *
+ * @param {Duration} duration
+ */
+export const getPastDate = (duration: Duration): number => sub(getCurrentJsDateObject(), duration).valueOf();
+
+// isExists uses 0 indexed month numbers
+/**
+ * Checks if a timetype is valid
+ * @param {DateType} date
+ */
+export const isValidDate = (date?: DateType): boolean => {
+  if (date) {
+    return isExists(date.year, date.month - 1, date.day);
+  }
+  return false;
+};
+
+/**
+ * Checks if a timetype is valid
+ * @param {TimeType} time
+ */
+export const isValidTime = (time?: TimeType): boolean => {
+  if (time && time.hours >= 0 && time.hours <= 23 && time.minutes >= 0 && time.minutes <= 59) {
+    return true;
+  }
+  return false;
+};
+
+/** Checks if a timezone string is valid
+* @param {string} timezone
+*/
+export const isValidTimeZone = (timezone: string): boolean => {
+  try {
+    utcToZonedTime(getCurrentJsDateObject(), timezone);
+    return true;
+  } catch {
+    return false;
+  }
+};
+
+export const zeroFill = (num: string, length: number): string => (num.toString().padStart(length, '0'));
+
+export const convertDateTypeToString = (date: DateType): string => {
+  const day = zeroFill(date.day.toString(), 2);
+  const month = zeroFill(date.month.toString(), 2);
+  const year = zeroFill(date.year.toString(), 4);
+
+  return `${year}-${month}-${day}`;
+};
+
+export const convertTimeTypeToString = (time: TimeType): string => {
+  const hours = zeroFill(time.hours.toString(), 2);
+  const minutes = zeroFill(time.minutes.toString(), 2);
+
+  return `${hours}:${minutes}`;
+};
+
+export const dateTypeToStandardEUDateFormat = (date: DateType): string => {
+  const day = zeroFill(date.day.toString(), 2);
+  const month = zeroFill(date.month.toString(), 2);
+  const year = zeroFill(date.year.toString(), 4);
+
+  return `${day}.${month}.${year}`;
+};
```

Comment:
Remarks about the function `dateTypeToStandardEUDateFormat`

1. This function format a string
2. This function depends on another function called `zeroFill`.
3. Naming

**My advices**

1. Use an established parser for that, like this
```
 const customDateFormat = (date: string, pattern: string): string => {
    const parsed = parse(date, pattern, new Date());
    return format(parsed,  pattern);
  };
  
// usage 
const dateType = { day: 22, month: 7, year: 2021 };
const formatedEU = customDateFormat( `${dateType.day}.${dateType.month}.${dateType.year}`, 'dd.MM.yyyy' ));
const formatedUS = customDateFormat( `${dateType.year}.${dateType.month}.${dateType.day}, 'yyyy-MM-dd' ));
```


2. Use the `Dependency Injection` pattern to pass a required function in the body of the main function.

3. `Naming` is important for your code reviewer and reader. `customDateFormat` might be easier to read.

4. Do apply the `DRY` principle

Reply:
Ended up removing the zerofill function and using a `customDateFormat` function